plugins/extractors/service_yaml/build_asset.go

```diff
@@ -0,0 +1,130 @@
+package serviceyaml
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/odpf/meteor/models"
+	v1beta2 "github.com/odpf/meteor/models/odpf/assets/v1beta2"
+	"github.com/odpf/meteor/plugins"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	service = "service_yaml"
+	typ     = "service"
+)
+
+func buildAsset(scope string, svc Service) (*v1beta2.Asset, error) {
+	upstreams, downstreams, err := buildLineage(svc)
+	if err != nil {
+		return nil, fmt.Errorf("build asset: %w", err)
+	}
+
+	data, err := anypb.New(&v1beta2.Service{
+		ApplicationId: svc.ID,
+		Version:       svc.Version,
+		CreateTime:    toTimestamp(svc.CreateTime),
+		UpdateTime:    toTimestamp(svc.UpdateTime),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("build asset metadata: %w", err)
+	}
+
+	var owners []*v1beta2.Owner
+	if svc.Team.ID != "" {
+		owners = append(owners, &v1beta2.Owner{
+			Urn:   svc.Team.ID,
+			Name:  svc.Team.Name,
+			Email: svc.Team.Email,
+		})
+	}
+
+	return &v1beta2.Asset{
+		Urn:         models.NewURN(service, scope, typ, svc.Name),
+		Name:        svc.Name,
+		Service:     service,
+		Type:        typ,
+		Url:         svc.URL,
+		Description: svc.Description,
+		Data:        data,
+		Owners:      owners,
+		Lineage: &v1beta2.Lineage{
+			Upstreams:   upstreams,
+			Downstreams: downstreams,
+		},
+		Labels: svc.Labels,
+	}, nil
+}
+
+func buildLineage(svc Service) (upstreams []*v1beta2.Resource, downstreams []*v1beta2.Resource, err error) {
+	upstreams, err = buildLineageResources(svc.Inputs)
+	if err != nil {
+		return nil, nil, fmt.Errorf("build lineage upstreams: %w", err)
+	}
+
+	downstreams, err = buildLineageResources(svc.Outputs)
+	if err != nil {
+		return nil, nil, fmt.Errorf("build lineage downstreams: %w", err)
+	}
+
+	return upstreams, downstreams, nil
+}
+
+func buildLineageResources(dd []Dataset) ([]*v1beta2.Resource, error) {
```

Comment:
@sudo-suhas should this be kept outside of the extractor? I assume this can go as part of the processor? I think this is a very specific usecase.

Reply:
Not sure if that is possible. Where would we store the dataset info on the `Asset` so that the processor can access it for populating lineage? Also, I don't see it as being different from what we are doing in other extractors. We parse the data sources/sinks to construct lineage.

Reply:
@ravisuhag judging from the parameter (`dd []Dataset`), it looks like this function is pretty specific to this processor.

We could generalise some of the logic so that other processors can use it but IMHO I think we can do that when the needs come.